pkg/spoke/addon/lease_controller.go

```diff
@@ -0,0 +1,191 @@
+package addon
+
+import (
+	"context"
+	"time"
+
+	addonv1alpha1 "github.com/open-cluster-management/api/addon/v1alpha1"
+	addonclient "github.com/open-cluster-management/api/client/addon/clientset/versioned"
+	addoninformerv1alpha1 "github.com/open-cluster-management/api/client/addon/informers/externalversions/addon/v1alpha1"
+	addonlisterv1alpha1 "github.com/open-cluster-management/api/client/addon/listers/addon/v1alpha1"
+	"github.com/open-cluster-management/registration/pkg/helpers"
+
```

Comment:
nit: we used to separate these blocks, but I honestly haven't found much value to separation beyond standard lib and the rest.  You could consider in the project whether you still see value in separating import blocks.